### PR TITLE
Use endpoints class

### PR DESCRIPTION
**Pull request by aljones15**
This is clean up: it removes tons of unneeded stuff from the actual test suite and instead creates an Endpoints class that really helps make the suite cleaner and makes it almost more inline with the rest of the suites. It also moves the challenge generation to `data-generator.js` further decluttering the test file.

### CODE REVIEW

File: tests/TestEndpoints.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import {proveVP} from './data-generator.js';
+import receiveJson from './receive-json.js';
+
+export class TestEndpoints {
+  constructor({implementation, tag}) {
+    this.implementation = implementation;
+    this.tag = tag;
+    this.verifier = implementation.verifiers.find(
+      verifier => verifier.tags.has(tag));
+    this. issuer = implementation.issuers.find(
```
Comment (BigBlueHat) [inline tests/TestEndpoints.js:18]:
```suggestion
    this.issuer = implementation.issuers.find(
```

**Review by BigBlueHat — CHANGES_REQUESTED** [review]
Tiny tweak. Otherwise looks good!

File: tests/TestEndpoints.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright (c) 2022-2023 Digital Bazaar, Inc. All rights reserved.
```
Comment (JSAssassin) [inline tests/TestEndpoints.js:2]:
```suggestion
 * Copyright (c) 2024 Digital Bazaar, Inc. All rights reserved.
```

**Review by JSAssassin — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -217,8 +217,8 @@ checked):
 Implementations are expected to not error when any of the following context
 files are used in a verifiable credential or a verifiable presentation:
 
-- [VC 2.0 Context - https://www.w3.org/ns/credentials/v2](https://www.w3.org/ns/credentials/v2)
-- [VC Examples Context - https://www.w3.org/ns/credentials/examples/v2](https://www.w3.org/ns/credentials/examples/v2)
+- [Vc 2.0 Context - https://www.w3.org/ns/credentials/v2](https://www.w3.org/ns/credentials/v2)
+- [Vc Examples Context - https://www.w3.org/ns/credentials/examples/v2](https://www.w3.org/ns/credentials/examples/v2)
```
Comment (BigBlueHat) [inline README.md:221]: Well...this takes the code style into the wrong place. 😃 These are for people, who typically abbreviate Verifiable Credentials as VC...and would read `Vc` as a typo. So, the README needs refixing.

File: tests/data-generator.js [hunk 4]
```diff
@@ -51,16 +56,16 @@ async function getKeys() {
 }
 
 /**
- * An extremely basic VP prover. This is not final
+ * An extremely basic Vp creator. This is not final
```
Comment (BigBlueHat) [inline tests/data-generator.js:59]: Again, these are human facing abbreviations, so need to stay all caps. Maybe a global search and replace got too greedy? 😀

Comment (BigBlueHat) [reply tests/data-generator.js:59]:
```suggestion
 * An extremely basic VP creator. This is not final
```

**Review by BigBlueHat — CHANGES_REQUESTED** [review]
Code styles don't apply to other abbreviations, fwiw.

File: tests/data-generator.js (hunk 4 above)
Comment (aljones15) [reply tests/data-generator.js]: I was actually just being thorough anyways here ya go: https://github.com/w3c/vc-data-model-2.0-test-suite/pull/60/commits/eda2a3b473cab756fa890aed271cab870fdf55db

File: README.md (hunk 3 above)
Comment (aljones15) [reply README.md]: This should make the corrections asked for: https://github.com/w3c/vc-data-model-2.0-test-suite/commit/eda2a3b473cab756fa890aed271cab870fdf55db

**Review by BigBlueHat — APPROVED** [review]

**Pull request merged**